MediaGalleryMetadata/Model/AddIptcMetadata.php

```diff
@@ -20,18 +23,48 @@ class AddIptcMetadata
     private const IPTC_DESCRIPTION_SEGMENT = '2#120';
     private const IPTC_KEYWORDS_SEGMENT = '2#025';
 
+    /**
+     * @var DriverInterface
+     */
+    private $driver;
+
+    /**
+     * @var FileReader
+     */
+    private $fileReader;
+
+    /**
+     * @var FileInterfaceFactory
+     */
+    private $fileFactory;
+
+    /**
+     * @param FileInterfaceFactory $fileFactory
+     * @param DriverInterface $driver
+     * @param FileReader $fileReader
+     */
+    public function __construct(
+        FileInterfaceFactory $fileFactory,
+        DriverInterface $driver,
+        FileReader $fileReader
+    ) {
+        $this->fileFactory = $fileFactory;
+        $this->driver = $driver;
+        $this->fileReader = $fileReader;
+    }
+    
     /**
      * Write metadata
      *
      * @param FileInterface $file
      * @param MetadataInterface $metadata
-     * @param SegmentInterface $segment
-     * @return string
+     * @param null|SegmentInterface $segment
      */
-    public function execute(FileInterface $file, MetadataInterface $metadata, SegmentInterface $segment): string
+    public function execute(FileInterface $file, MetadataInterface $metadata, ?SegmentInterface $segment): FileInterface
     {
-        if (is_callable('iptcembed')) {
-            $iptcData = iptcparse($segment->getData());
+        if (is_callable('iptcembed') && is_callable('iptcparse')) {
```

Comment:
Let's throw an exception if functions are not callable

Reply:
ok